Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp

```diff
@@ -668,6 +669,22 @@ static void autoActivateFont(const String& name, CGFloat size)
 
 std::unique_ptr<FontPlatformData> FontCache::createFontPlatformData(const FontDescription& fontDescription, const AtomString& family, const FontCreationContext& fontCreationContext, OptionSet<FontLookupOptions> options)
 {
+    if (auto& fontMap = userInstalledFontMap()) {
+        auto fontKey = family.string().convertToASCIILowercase();
+        if (auto fontURL = fontMap->getOptional(fontKey)) {
+            RELEASE_LOG(Fonts, "FontCache user installed fonts contains family %{private}s with fontURL %{private}s", family.string().utf8().data(), fontURL->string().utf8().data());
+            RetainPtr cfURL = fontURL->createCFURL();
+
+            CFErrorRef error = nullptr;
+            if (!CTFontManagerRegisterFontsForURL(cfURL.get(), kCTFontManagerScopeProcess, &error)) {
+                RetainPtr descriptionCF = adoptCF(CFErrorCopyDescription(error));
+                String error(descriptionCF.get());
+                RELEASE_LOG(Fonts, "Could not register font family %{private}s, error %s", family.string().utf8().data(), error.utf8().data());
+            }
+            fontMap->remove(fontKey);
```

Comment:
If `CTFontManagerRegisterFontsForURL` is slow, should we also remove the keys with the same URL from `fontMap`?